### DISCUSSION TITLE
How to bring your own data in the traffic library?

### DISCUSSION THREAD

**Discussion by laynr** [xoolive/traffic#311] [Q&A] [answered]
I have the most basic generated time-series trajectory flight data in CSV files with `icao24`, `datetime`, `lat`, `lon`, and `alt`. What is the best way to bring it into the traffic library?

Looked for help at this page:  https://traffic-viz.github.io/user_guide/own_data.html

**Comment by xoolive** [comment, marked as answer]
Hi @laynr 
Indeed, I am well behind in terms of documentation.

About your issue:
- rename `lat`, `lon` and `alt` to `latitude`, `longitude`, `altitude`
- add a column `timestamp` with a dtype as timestamp (I recommend the timezone-aware version, but things **should** work in both cases)

Then create a `Flight` or `Traffic` (collection of `Flight`) object directly from the pandas Dataframe:

```python
# for a single trajectory
Flight(df.rename(columns=dict(lat='latitude', etc.)))
# for many trajectories
for flight in Traffic(df.rename(columns=dict(lat='latitude', etc.))):
    print(flight)
```
